File: Java/src/main/java/airports/Airport.java
package airports;

import planes.ExperimentalPlane;
import models.MilitaryType;
import planes.MilitaryPlane;
import planes.PassengerPlane;
import planes.Plane;
import java.util.Comparator;
import java.util.List;
import java.util.NoSuchElementException;
import java.util.stream.Collectors;

public class Airport {
    private List<? extends Plane> planes;

    public Airport(List<? extends Plane> planes) {
        this.planes = planes;
    }

    public List<? extends Plane> getPlanes() {
        return planes;
    }
    public <T extends Plane> List<T> getPlanesByType(Class<T> planeType) {
        return planes.stream()
                .filter(planeType::isInstance)
                .map(planeType::cast)
                .collect(Collectors.toList());
    }

    public List<PassengerPlane> getPassengerPlanes() {
        return getPlanesByType(PassengerPlane.class);
    }


    public List<MilitaryPlane> getMilitaryPlanes() {
        return getPlanesByType(MilitaryPlane.class);
    }

    public List<ExperimentalPlane> getExperimentalPlanes() {
        return getPlanesByType(ExperimentalPlane.class);
    }

    public PassengerPlane getPassengerPlaneWithMaxPassengersCapacity() {
        return getPassengerPlanes().stream()
                .max(Comparator.comparingInt(PassengerPlane::getPassengersCapacity))
                .orElseThrow(() -> new NoSuchElementException("There are not passenger planes"));
    }

    public List<MilitaryPlane> getMilitaryPlanesByType(MilitaryType type) {
        return getMilitaryPlanes().stream()
                .filter(plane -> plane.getType() == type)
                .collect(Collectors.toList());
    }

    public List<MilitaryPlane> getTransportMilitaryPlanes() {
        return getMilitaryPlanesByType(MilitaryType.TRANSPORT);
    }

    public List<MilitaryPlane> getBomberMilitaryPlanes() {
        return getMilitaryPlanesByType(MilitaryType.BOMBER);
    }

    public Airport sortBy(Comparator<Plane> comparator) {
        planes.sort(comparator);
        return this;
    }

    public Airport sortByMaxDistance() {
        return sortBy(Comparator.comparingInt(Plane::getMaxFlightDistance));
    }

    public Airport sortByMaxSpeed() {
        return sortBy(Comparator.comparingInt(Plane::getMaxSpeed));
    }

    public Airport sortByMaxLoadCapacity() {
        return sortBy(Comparator.comparingInt(Plane::getMaxLoadCapacity));
    }

    @Override
    public String toString() {
        return "airports.Airport{" +
                "planes=" + planes.toString() +
                '}';
    }

}

File: Net/Aircompany/Airport.cs
using Aircompany.Models;
using Aircompany.Planes;
using System.Collections.Generic;
using System.Linq;

namespace Aircompany;

public class Airport(IEnumerable<Plane> planes)
{
    private List<Plane> Planes = planes.ToList();

    public List<PassengerPlane> GetPassengersPlanes()
    {
        return Planes.Where(t => t.GetType() == typeof(PassengerPlane)).Cast<PassengerPlane>().ToList();
    }

    public List<MilitaryPlane> GetMilitaryPlanes()
    {
        return Planes.Where(t => t.GetType() == typeof(MilitaryPlane)).Cast<MilitaryPlane>().ToList();
    }

    public PassengerPlane GetPassengerPlaneWithMaxPassengersCapacity()
    {
        var passengerPlanes = GetPassengersPlanes();
        return passengerPlanes.Aggregate((w, x) => w.PassengersCapacityIs() > x.PassengersCapacityIs() ? w : x);             
    }

    public List<MilitaryPlane> GetTransportMilitaryPlanes()
    {
        var militaryPlanes = GetMilitaryPlanes();

        return militaryPlanes.Where(plane => plane.PlaneTypeIs() == MilitaryType.TRANSPORT).ToList();
    }

    public Airport SortByMaxDistance()
    {
        return new Airport(Planes.OrderBy(w => w.MaxFlightDistance()));
    }

    public Airport SortByMaxSpeed()
    {
        return new Airport(Planes.OrderBy(w => w.GetMaxSpeed()));
    }

    public Airport SortByMaxLoadCapacity()
    {
        return new Airport(Planes.OrderBy(w => w.MaxLoadCapacity()));
    }


    public IEnumerable<Plane> GetPlanes()
    {
        return Planes;
    }

    public override string ToString()
    {
        return "Airport{" +
               "planes=" + string.Join(", ", Planes.Select(x => x.GetModel())) +
               '}';
    }
}